src/java.base/share/classes/java/lang/StableValue.java

```diff
@@ -0,0 +1,747 @@
+ * <p>
+ * The fibonacci example above is a dependency graph with no circular dependencies (i.e.,
+ * it is a dependency tree):
+ *{@snippet lang=text :
+ *
+ *              ___________fib(5)____________
+ *             /                             \
+ *       ____fib(4)____                  ____fib(3)____
+ *      /              \                /              \
+ *    fib(3)          fib(2)          fib(2)          fib(1)
+ *   /     \         /     \         /     \
+ * fib(2) fib(1)   fib(1) fib(0)   fib(1) fib(0)
+ *}
+ *
+ * If there are circular dependencies in a dependency graph, a stable value will
+ * eventually throw an {@linkplain IllegalStateException} upon referencing elements in
+ * a circularity.
+ *
+ * <h2 id="thread-safety">Thread Safety</h2>
+ * The content of a stable value is guaranteed to be set at most once. If competing
+ * threads are racing to set a stable value, only one update succeeds, while other updates
+ * are blocked until the stable value becomes set.
+ * <p>
+ * The at-most-once write operation on a stable value that succeeds
+ * (e.g. {@linkplain #trySet(Object) trySet()})
+ * {@linkplain java.util.concurrent##MemoryVisibility <em>happens-before</em>}
+ * any subsequent read operation (e.g. {@linkplain #orElseThrow()}) that is successful.
+ * A successful write operation can be either:
+ * <ul>
+ *     <li>a {@link #trySet(Object)} that returns {@code true},</li>
+ *     <li>a {@link #setOrThrow(Object)} that does not throw, or</li>
+ *     <li>an {@link #orElseSet(Supplier)} that successfully runs the supplier</li>
+ * </ul>
+ * A successful read operation can be either:
+ * <ul>
+ *     <li>a {@link #orElseThrow()} that does not throw,</li>
+ *     <li>a {@link #orElse(Object) orElse(other)} that does not return the {@code other} value</li>
+ *     <li>an {@link #orElseSet(Supplier)} that does not {@code throw}, or</li>
+ *     <li>an {@link #isSet()} that returns {@code true}</li>
+ * </ul>
+ * <p>
+ * The method {@link #orElseSet(Supplier)} guarantees that the provided
+ * {@linkplain Supplier} is invoked successfully at most once even under race.
+ * Invocations of {@link #setOrThrow(Object)} forms a total order of zero or more
+ * exceptional invocations followed by zero (if the content was already set) or one
+ * successful invocation. Since stable functions and stable collections are built on top
+ * of {@linkplain StableValue#orElseSet(Supplier) orElseSet()} they too are
+ * thread safe and guarantee at-most-once-per-input invocation.
+ *
+ * <h2 id="performance">Performance</h2>
+ * The _content_ of a set stable value is treated as a constant by the JVM, provided that
+ * the reference to the stable value is also constant (e.g. in cases where the
+ * stable value itself is stored in a {@code static final} field). Stable functions and
+ * collections are built on top of StableValue. As such, they are also treated as
+ * constants by the JFM.
+ * <p>
```

Comment:
JFM -> JVM

Reply:
German pronunciation ...  Jokes aside, thanks for spotting this typo.